File: backend/src/schemas/sharing.py

from pydantic import BaseModel, EmailStr
import uuid
from typing import List
from datetime import datetime


class ShareDocumentRequest(BaseModel):
    user_ids: List[uuid.UUID]


class CollaboratorInfo(BaseModel):
    user_id: uuid.UUID
    username: str
    email: EmailStr
    access_type: str
    granted_at: datetime

    class Config:
        from_attributes = True


class ShareResponse(BaseModel):
    document_id: uuid.UUID
    shared_with: List[CollaboratorInfo]


class CollaboratorsResponse(BaseModel):
    document_id: uuid.UUID
    collaborators: List[CollaboratorInfo]

File: frontend/src/types/sharing.ts
export interface CollaboratorInfo {
  user_id: string;
  username: string;
  email: string;
  access_type: 'owner' | 'editor';
  granted_at: string;
}

export interface ShareDocumentRequest {
  user_ids: string[];
}

export interface ShareResponse {
  document_id: string;
  shared_with: CollaboratorInfo[];
}

export interface CollaboratorsResponse {
  document_id: string;
  collaborators: CollaboratorInfo[];
}

export interface UserSearchResult {
  id: string;
  username: string;
  email: string;
}
